src/awkward/index.py

```diff
@@ -214,6 +215,9 @@ def form(self) -> str:
         return _dtype_to_form[self._data.dtype]
 
     def __getitem__(self, where):
+        if isinstance(where, slice):
+            where = normalize_slice(where, nplike=self.nplike)
```

Comment:
We need to normalize the slice so that `unknown_length` becomes an unknown scalar.